### ISSUE TITLE
[Support] How to append custom attribute to array in LDAPRecord Model?

### ISSUE THREAD

**Issue by satyakresna** [DirectoryTree/LdapRecord#267]
Hi @stevebauman, is that possible to append custom attribute to array in LDAPRecord model like Laravel has (Laravel Eloquent: Serialization: [Appending Values To JSON](https://laravel.com/docs/8.x/eloquent-serialization#appending-values-to-json))?

Here's the custom attribute I create where `domainname` attribute:

```php

<?php

namespace App\Ldap;

use LdapRecord\Models\Model;

class User extends Model
{

    public function getDomainNameAttribute()
    {
        return $this->getDn();
    }
}
```

I expect `domainname` attribute added to attributes array.

![result](https://user-images.githubusercontent.com/11099186/115344442-04c99c80-a1e0-11eb-8001-34a096afa380.PNG)

**Comment by stevebauman** [comment]
Hi @satyakresna!

Great question! Unfortunately accessors don't get appended to the arrayable attributes right now (there is no `$appends` feature). Though you've just pin-pointed a must-have feature that I have to port over from Eloquent. I'll definitely be adding this in 👍 

To get this done right now in your application, you can override the `getArrayableAttributes()` method on your `User` model and add the `domain_name` attribute to the array:

```php
<?php

namespace App\Ldap;

use LdapRecord\Models\Model;

class User extends Model
{
    public function getDomainNameAttribute()
    {
        return $this->getDn();
    }

    protected function getArrayableAttributes()
    {
        return array_merge(parent::getArrayableAttributes(), [
            'domain_name' => $this->getDomainNameAttribute(),
        ]);
    }
}
```

Give me a day or two to implement this so it's done automatically for you, just as it's done in Eloquent!

**Comment by satyakresna** [comment]
That's great! Thanks Steve! For now I'm using your solution and it works.

**Comment by stevebauman** [comment]
Glad you're up and running! 😄 

Okay, I've just released [v2.4.3](https://github.com/DirectoryTree/LdapRecord/releases/tag/v2.4.3) that contains the ported `$appends` feature 👍 

There is a difference to note however: `PascalCased` accessors must be defined in their *hyphenated* format in the `$appends` array, and will also appear in their hyphenated format in the resulting JSON string. For example, in your case above:

```php
<?php

namespace App\Ldap;

use LdapRecord\Models\Model;

class User extends Model
{
    protected $appends = ['domain-name'];

    public function getDomainNameAttribute()
    {
        return $this->getDn();
    }
}
```

```php
// Displays: "cn=john doe,dc=local,dc=com"
echo $user->domain_name;

// Displays: "cn=john doe,dc=local,dc=com"
echo $user->getFirstAttribute('domain-name');

// Displays: "{"domain-name":["cn=john doe,dc=local,dc=com"]}"
echo json_encode($user);
```

This is because LDAP does not actually support attributes with underscores, so to stay true to LDAP's attribute implementation and to avoid confusion, this is how it will work.

Hope this helps! Thanks for the feature request!

**Comment by satyakresna** [comment]
Hi @stevebauman, it works great!
